### PR TITLE
Fix WASM ABI for read_db, write_db, canonicalize_address, humanize_address + Implement remove_db

### LINKED ISSUE #276: Implement remove_db

### LINKED ISSUE #278: Modify current contract operations for 0.9
- [x] read_db
 - [x] write_db
 - [x] humanize_address
 - [x] canonicalize_address

### PR DESCRIPTION
**Pull request by assafmo**
Fixes #278 
Fixes #276

### CODE REVIEW

**Comment by assafmo** [comment]
Depends on #300, I think

File: cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs [hunk 1]
```diff
@@ -11,19 +11,34 @@ pub enum WasmEngineError {
...
+    DbError(DbError),
+    MemoryAllocationError,
+    MemoryReadError,
+    MemoryWriteError,
+    InputInvalid,
+    InputEmpty,
+    InputWrongPrefix,
+    InputWrongLength,
+    OutputWrongLength,
+    NonExistentImportFunction,
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs:23]: please move these variants to their own enum like `DbError`, called `CommunicationError` or `ContractAbiError`

Comment (assafmo) [reply cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs:23]: Why?? Seems to me like wrapping for the sake of wrapping

Comment (reuvenpo) [reply cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs:23]: well, sort of.
it's mostly to keep things organized, much like how DbError was implemented.

**Pull request merged**